src/main/scala/quasar/sbtdatasource/AssemblePlugin.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.sbtdatasource
+
+import scala.{Array, List, StringContext}
+import scala.Predef.{ArrowAssoc, String, genericWrapArray, println}
+import scala.collection.Seq
+
+import java.io.{File, IOException}
+import java.lang.{Exception, Runtime, SuppressWarnings}
+import java.nio.file.{Path, Files, SimpleFileVisitor, FileVisitResult}
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+import java.nio.file.StandardOpenOption.CREATE_NEW
+
+import cats.Parallel
+import cats.data.ValidatedNel
+import cats.effect.Sync
+import cats.instances.list._
+import cats.instances.string._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.parallel._
+import cats.syntax.traverse._
+
+import coursier._
+import coursier.util.Schedulable
+
+import io.circe.Json
+
+import sbt.ModuleID
+
+/** Assemble a datasource plugin tarball, returning the path to the artifact. */
+object AssemblePlugin {
```

Comment:
This was copied from slamdata/quasar-s3 and updated to use the newer typeclasses.